src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs

```diff
@@ -309,6 +312,26 @@ public bool TryCopyTo(Span<T> destination)
             return retVal;
         }
 
+        /// <summary>
+        /// If <paramref name="index"/> is within the span, dereferences the element at
+        /// that index, outs it via <paramref name="value"/>, and returns <see langword="true"/>.
+        /// Otherwise returns <see langword="false"/>.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal unsafe bool TryGetElementAt(nuint index, [MaybeNullWhen(false)] out T value)
```

Comment:
I am not fan of improving performance by internal helpers like this one that are not available in public surface. This should be either turned into public API, or we should live without it.

Reply:
@jkotas I understand the concern. How about I experiment in a separate branch with plumbing it through other call sites and see what the perf impact is? That should give us evidence that this API is (or isn't) generally worthwhile.